### PR TITLE
Move pipeline selection inversion logic (for personalization widget) to the UI

### PR DESCRIPTION
**Pull request by marques-work**

### CODE REVIEW

**Comment by ketan** [comment]
LGTM, can you squash, rebase on latest and run the PR build?

**Comment by ibnc** [comment]
@ketan done. Will run the PR build is running now

**Comment by marques-work** [comment]
@ketan build-linux passed, but windows agents seem to be having trouble. same seen on master.

File: server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js [hunk 1]
```diff
@@ -37,14 +37,14 @@ const PipelineSelection = function (pipelineGroups, selections, blacklist) {
     const selections   = [];
 
     _.each(allPipelines, (selection, pipelineName) => {
-      if (selection()) {
+      if (self.blacklist() ^ selection()) {
```
Comment (marques-work) [inline server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js]: Inversion now happens here

File: server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js [hunk 2]
```diff
@@ -58,14 +58,14 @@ const PipelineSelection = function (pipelineGroups, selections, blacklist) {
 PipelineSelection.fromJSON = (json) => {
   const pipelineGroups = json.pipelines;
   const selections     = {};
-
+  const blacklist      = json.blacklist;
```
Comment (marques-work) [inline server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js]: and here,

File: server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js [hunk 3]
```diff
@@ -58,14 +58,14 @@ const PipelineSelection = function (pipelineGroups, selections, blacklist) {
...
   _.each(_.keys(pipelineGroups), (group) => {
     _.each(pipelineGroups[group], (pipeline) => {
-      selections[pipeline] = Stream(_.includes(json.selections, pipeline));
+      selections[pipeline] = Stream(!!(blacklist ^ _.includes(json.selections, pipeline)));
```
Comment (marques-work) [inline server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js]: and here.

File: api/api-pipeline-selection-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineselection/representers/PipelineSelectionResponse.java [hunk 4]
```diff
@@ -17,21 +17,26 @@
...
 
-    public PipelineSelections getSelectedPipelines() {
-        return selectedPipelines;
+    public List<String> selections() {
+        return pipelines;
+    }
+
+    public boolean blacklist() {
+        return blacklist;
     }
```
Comment (marques-work) [inline api/api-pipeline-selection-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineselection/representers/PipelineSelectionResponse.java]: Avoid needing to construct a `PipelineSelections` instance just to be able to de/serialize. Makes testing easier, and the usage clearer during controller actions.

File: api/api-pipeline-selection-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineselection/representers/PipelineSelectionResponse.java [hunk 5]
```diff
@@ -17,21 +17,26 @@
...
 
-    public PipelineSelections getSelectedPipelines() {
-        return selectedPipelines;
+    public List<String> selections() {
+        return pipelines;
+    }
+
+    public boolean blacklist() {
+        return blacklist;
```
Comment (marques-work) [inline api/api-pipeline-selection-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineselection/representers/PipelineSelectionResponse.java]: Avoid needing to construct a `PipelineSelections` instance just to be able to de/serialize. Makes testing easier, and the usage clearer during controller actions.

File: server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js (hunk 3 above)
Comment (marques-work) [reply server/webapp/WEB-INF/rails.new/webpack/models/dashboard/pipeline_selection.js]: and yes, that's an XOR.

**Pull request merged**